x/tally/types/hash_sort.go

```diff
@@ -0,0 +1,47 @@
+package types
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"slices"
+)
+
+type HashSortable interface {
+	GetSortKey() []byte
+}
+
+func HashSort[T HashSortable](originalItems []T, entropy ...[]byte) []T {
+	type sortItem struct {
+		sortKey  []byte
+		original T
+	}
+	totalItems := len(originalItems)
+
+	sortItems := make([]sortItem, totalItems)
+
+	hasher := sha256.New()
+	for i := range totalItems {
+		hasher.Reset()
+
+		hasher.Write(originalItems[i].GetSortKey())
+		for _, entropy := range entropy {
+			hasher.Write(entropy)
+		}
```

Comment:
It's confusing that the iterator and the iteration list have the same name here.

Reply:
Why are we iterating through entropy here?